tests/test_heartbeats.cpp

```diff
@@ -370,13 +370,19 @@ void test_setsockopt_heartbeat_ttl_max ()
 
 void test_setsockopt_heartbeat_ttl_more_than_max_fails ()
 {
+#if !defined(ZMQ_ACT_MILITANT)
     void *const socket = test_context_socket (ZMQ_PAIR);
     const int value = heartbeat_ttl_max + 1;
     TEST_ASSERT_FAILURE_ERRNO (
       EINVAL,
       zmq_setsockopt (socket, ZMQ_HEARTBEAT_TTL, &value, sizeof (value)));
 
     test_context_socket_close (socket);
+#else
+    TEST_IGNORE_MESSAGE (
+      "libzmq with ZMQ_ACT_MILITANT, ignoring "
```

Comment:
Why?

Reply:
ZMQ_ACT_MILITANT causes the test program to voluntarily crash on invalid options, which causes the test to fail erroneously.

Reply:
then don't enable it

Reply:
I think the “militant” mode has great value. Maybe it should be renamed “strict” or something and have its use much expanded in the codebase so the library bugchecks on every programmer error (like passing invalid values to a function) instead of returning error codes in those cases. 

What is not OK is having tests fail (not in the usual sense of an expected failure) with the strict mode enabled. 

Unless there is a way to express that crashing is a form of success (and then those tests should be standalone as nothing runs after them) the invalid parameter tests must be disabled in strict mode.

Reply:
No, this is fine as it is. Drop this change and the rest of the PR can be merged.